File: agc023/a/main.rb
require 'pp'
N = gets.chomp.to_i
a = gets.chomp.split(" ").map(&:to_i)

ans = 0
hash = Hash.new(0)
sum = 0
N.times do |i|
  x = a[i]
  sum += x
  hash[x - sum] += 1
  ans += hash[-sum]
  # pp sum, hash
end

puts ans

File: agc023/a/main.go
package main

import (
	"fmt"
)

func main() {
	var (
		n int
	)
	fmt.Scan(&n)
	ii := make([]int, n)
	for i := 0; i < n; i++ {
		fmt.Scan(&ii[i])
	}

	ans := 0

	for i := 0; i < n; i++ {
		sum := 0
		for j := i; j < n; j++ {
			sum += ii[j]
			if sum == 0 {
				ans++
			}
		}
	}

	fmt.Printf("%d\n", ans)
}
